### PR TITLE
[script.module.archive_tool] 1.13

### PR DESCRIPTION
**Pull request by zach-morris**
https://github.com/zach-morris/script.module.archive_tool/tree/master/lib/archive_tool
Leia

### Description

archive_tool is a set of common python functions to work with the Kodi archive virtual file system (vfs) binary addons such as vfs.libarchive and vfs.rar
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.module.archive_tool/addon.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<addon id="script.module.archive_tool" name="archive_tool" provider-name="Zach Morris" version="1.13">
+  <requires>
+    <import addon="xbmc.python" version="2.26.0"/>
+    <import addon="script.module.kodi-six"/>
+    <import addon="vfs.libarchive" version="1.0.5"/>
+    <import addon="vfs.rar" version="2.3.0" optional="true"/>
+  </requires>
+  <extension point="xbmc.python.module" library="lib" />
+  <extension point="xbmc.addon.metadata">
+    <summary lang="en">Python class for working with file archives</summary>
```
Comment (enen92) [inline script.module.archive_tool/addon.xml:11]: en_GB

File: script.module.archive_tool/addon.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <description lang="en">archive_tool is a set of common tools to work with the Kodi libarchive virtual file system</description>
```
Comment (enen92) [inline script.module.archive_tool/addon.xml:12]: en_GB

File: script.module.archive_tool/addon.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <platform>all</platform>
+    <assets>
+      <icon>icon.png</icon>
+    </assets>
+    <license>free</license>
```
Comment (enen92) [inline script.module.archive_tool/addon.xml:17]: Such license doesn't exist. According to the License of your addon this should be `GPL-3.0-only` (spdx identifier)

File: script.module.archive_tool/lib/archive_tool/archive_tool.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+from kodi_six import xbmc, xbmcvfs
+try:
+	from urllib import quote_plus as url_quote #Python 2
+except:
+	from urllib.parse import quote_plus as url_quote #Python 3
+import os, json
+
+class archive_tool(object):
+
+	def __init__(self, archive_file=None, files_to_extract=None, directory_out=None,use_vfs_rar=False):
```
Comment (enen92) [inline script.module.archive_tool/lib/archive_tool/archive_tool.py:46]: I'd suggest you run reindent in your codebase

File: script.module.archive_tool/lib/archive_tool/run_archive_tool_test.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+from kodi_six import xbmc, xbmcaddon, xbmcgui, xbmcvfs
```
Comment (enen92) [inline script.module.archive_tool/lib/archive_tool/run_archive_tool_test.py:1]: if this is a test file (and not used) please remove from the submission

File: script.module.archive_tool/addon.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.module.archive_tool" name="archive_tool" provider-name="Zach Morris" version="1.13">
```
Comment (enen92) [inline script.module.archive_tool/addon.xml:2]: please use `major.minor.revision` as version scheme

**Comment by zach-morris** [comment]
Thanks, I will resolve these.  Do I fix this commit or delete it and push a new one?

**Comment by enen92** [comment]
Just add commits to it and I'll squash in the end. We're working on ways to automate all this.

**Comment by zach-morris** [comment]
Closing per other commit

**Pull request closed without merging**